eth/chains/base.py

```diff
@@ -164,7 +176,7 @@ def get_vm_class(cls, header: BlockHeader) -> Type['BaseVM']:
         return cls.get_vm_class_for_block_number(header.block_number)
 
     @abstractmethod
-    def get_vm(self, header: BlockHeader=None) -> 'BaseVM':
+    def get_vm(self, header: BlockHeader=None) -> 'VM':
```

Comment:
If you create an abstract method `apply_all_transactions` in `BaseVM` you should be able to keep these here on `BaseVM` rather than `VM`